dowhy/causal_refuter.py

```diff
@@ -38,70 +38,124 @@ def get_estimator_object(new_data, identified_estimand, estimate):
+
         Null Hypothesis: The estimate is a part of the distribution
         Alternative Hypothesis: The estimate does not fall in the distribution.
 
+        Parameters
+        -----------
         'estimate': CausalEstimate
         The estimate obtained from the estimator for the original data.
         'simulations': np.array
         An array containing the result of the refuter for the simulations
-        'distribution': string
-        The underlying distribution of the data
+        'test_type': string, default 'auto'
+        The type of test the user wishes to perform.
+        'significance_level': float, default 0.05
+        The significance level for the statistical test
+
+        Returns
+        --------
+        significance_dict: Dict
+        A Dict containing the p_value and a boolean that indicates if the result is statistically significant
         """
-        num_simulations = len(simulations)
-        if num_simulations > 200: # Bootstrapping
-            self.logger.info("Making use of Bootstrap as we have more than 200 examples.\n \
-             Note: The greater the number of examples, the more accurate are the confidence estimates")
-            # Sort the simulations
-            simulations.sort()
-            # Obtain the median value
-            median_refute_values= simulations[int(num_simulations)/2]
-
-            # Performing a two sided test
-            if estimate.value > median_refute_values:
-                # np.searchsorted tells us the index if it were a part of the array
-                # We select side to be left as we want to find the first value that matches
-                estimate_index = np.searchsorted(simulations, estimate.value, side="left")
-                # We subtact 1 as we are finding the value from the right tail
-                p_value = 1 - (estimate_index/ num_simulations)
+        # Initializing the p_value
+        p_value = 0
+
+        if test_type == 'auto':
+            num_simulations = len(simulations)
+            if num_simulations >= 100: # Bootstrapping
+                self.logger.info("Making use of Bootstrap as we have more than 200 examples.\n \
+                Note: The greater the number of examples, the more accurate are the confidence estimates")
+
+                # Perform Bootstrap Significance Test with the original estimate and the set of refutations
+                p_value = self.perform_bootstrap_test(estimate, simulations)
+
             else:
-                # We take the side to be right as we want to find the last index that matches
-                estimate_index = np.searchsorted(simulations, estimate.value, side="right")
-                # We get the probability with respect to the left tail.
-                p_value = estimate_index / num_simulations
+                self.logger.warn("We make use of the Normal Distribution as the sample has less than 200 examples.\n \
```

Comment:
same, you can change to 100. also better to say , "We assume", rather than "make use of"

Reply:
Done.